FormalConjectures/ErdosProblems/107.lean

```diff
@@ -0,0 +1,121 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 107
+
+*Reference:* [erdosproblems.com/107](https://www.erdosproblems.com/107)
+-/
+
+open Filter
+
+namespace Erdos107
+
+abbrev Point := EuclideanSpace ℝ (Fin 2)
+
+namespace Point
+
+/-- The three points `a,b,c` are not collinear. -/
+def InGenPos₃ (a b c : Point) : Prop :=
+  Matrix.det !![a 0, b 0, c 0 ; a 1, b 1, c 1 ; 1, 1, 1] ≠ 0
+
+/-- No three points in `S` are collinear. -/
+def SetInGenPos (S : Set Point) : Prop :=
+  ∀ ⦃p q r : Point⦄, {p,q,r} ⊆ S → List.Nodup [p,q,r] → InGenPos₃ p q r
+
+end Point
+
+/-- The set `S` is in convex position. -/
+def ConvexIndep (S : Set Point) : Prop :=
+  ∀ a ∈ S, a ∉ convexHull ℝ (S \ {a})
+
+/-- The set `P` contains a convex `n`-gon. -/
+def HasConvexNGon (n : ℕ) (P : Set Point) : Prop :=
+  ∃ S : Finset Point, S.card = n ∧ ↑S ⊆ P ∧ ConvexIndep S
+
+noncomputable def f (n : ℕ) : ℕ :=
```

Comment:
I have roughly followed the formalization of Erdős 126, except that [there](https://github.com/google-deepmind/formal-conjectures/blob/ed41ca10bdce5343dfe037abc84befc685c86016/FormalConjectures/ErdosProblems/126.lean#L54) the claims are stated as 'for any $f(n)$ matching the spec, ..', whereas here I have written down $f(n)$ and claimed that it has some properties. This is slightly stronger because it implies $f(n)$ exists and is not trivial (with mathlib's `sInf`, `f n = 0` if the set `{ N ∣ .. }` is infinite).

Reply:
one could make this implicit, by having another theorem showing that you set is nonempty (ne need to fill in the proof..)

Reply:
One could! Are you suggesting to implement that, or just pointing out another formulation?